File: python/BaekJoon/1000~9999/2000~2999/2884.py
A, B = map(int, input().split())

if B>=45 : B = B - 45
else :
    if A==0 :
        A = 23
        B = B + 15
    else :
        A = A - 1
        B = B + 15

print(A, B, end=' ')

File: js/BaekJoon/2884.js
const fs = require("fs");

//let [A, B] = fs.readFileSync("input.txt").toString().split(" ");
let [A, B] = fs.readFileSync("/dev/stdin").toString().split(" ");

A = Number(A);
B = Number(B);

if (B >= 45) {
  B = B - 45;
} else {
  if (A == 0) {
    A = 23;
    B = B + 15;
  } else {
    A = A - 1;
    B = B + 15;
  }
}

console.log(String(A) + " " + String(B));
